core/activity/src/dao/event.rs

```diff
@@ -99,65 +106,66 @@ impl<'c> EventDao<'c> {
 
     pub async fn get_events(
         &self,
-        identity_id: &str,
+        after_timestamp: DateTime<Utc>,
+        app_session_id: &Option<String>,
         max_events: Option<u32>,
     ) -> Result<Option<Vec<ProviderEvent>>> {
         use schema::activity::dsl;
         use schema::activity_event::dsl as dsl_event;
 
+        let app_session_id = app_session_id.to_owned();
         let limit = match max_events {
-            Some(val) => min(MAX_EVENTS, val),
+            Some(val) => MAX_EVENTS.min(val as i64),
             None => MAX_EVENTS,
         };
 
         log::trace!("get_events: starting db query");
-        let identity_id = identity_id.to_owned();
-
-        do_with_transaction(self.pool, move |conn| {
-            let results: Option<Vec<Event>> = dsl_event::activity_event
+        readonly_transaction(self.pool, move |conn| {
+            let mut query = dsl_event::activity_event
                 .inner_join(schema::activity::table)
-                .filter(dsl_event::identity_id.eq(identity_id))
                 .select((
                     dsl_event::id,
+                    dsl_event::event_date,
                     dsl_event::event_type_id,
                     dsl::natural_id,
                     dsl::agreement_id,
                     dsl_event::requestor_pub_key,
                 ))
+                .filter(dsl_event::event_date.gt(after_timestamp.naive_utc()))
+                .into_boxed();
+
+            if let Some(app_sid) = app_session_id {
+                query = query.filter(dsl_event::app_session_id.eq(app_sid));
+            }
+
+            let results: Option<Vec<Event>> = query
                 .order(dsl_event::event_date.asc())
-                .limit(limit as i64)
+                .limit(limit)
                 .load::<Event>(conn)
                 .optional()?;
 
-            if let Some(r) = &results {
-                let ids = r.iter().map(|event| event.id).collect::<Vec<_>>();
-                if !ids.is_empty() {
-                    diesel::delete(dsl_event::activity_event.filter(dsl_event::id.eq_any(ids)))
```

Comment:
we need some cleanup mechanism for those events

Reply:
Do you mean events in general? The count of activity events is going to be few orders of magnitude lower than payments / market

Reply:
I think like, at most one order of magnitude.
But yes, in general.
We have a cleanup mechanism in mkt

Reply:
| I think like, at most one order of magnitude.

E.g. debit note events may significantly outnumber create / destroy activity events for long running services.

| we need some cleanup mechanism for those events

We haven't established a common cleanup policy and event removal isn't a part of the spec (for whatever reason). I'll abstain from implementing a heuristic solution for now - there is simply no need to do that now and the number of events won't affect the database in a significant way

Reply:
True dat!
I was not clear in my intention here. I just wanted to express that such cleanup is needed, but not pushing to implement it right away. 👍

Reply:
referenced it #918